vertical-pod-autoscaler/enhancements/NNNN-max-allowed-recommendation/README.md

```diff
@@ -0,0 +1,45 @@
+# KEP-NNNN: Max allowed recommendation configurable
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+   - [Goals](#goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+   - [Test Plan](#test-plan)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+VPA currently allows configuring the maximum allowed recommendation per resource using the field `.spec.resourcePolicy.containerPolicies[].maxAllowed`. This helps ensure that recommendations don't exceed the allocatable capacity of the largest node. Without this limit, a high recommendation could make the Pod unschedulable, causing downtime. We aim to improve the VPA recommender by making it aware of the maximum node size, preventing it from generating recommendations that would render Pods unschedulable.
+
+## Motivation
+In dynamic environments, manual configuration of .maxAllowed can be error-prone, especially as cluster sizes and node configurations change over time. Automating this process by making the VPA recommender aware of the largest node size ensures more reliable scaling decisions. This prevents situations where Pods become unschedulable due to excessive resource recommendations.
```

Comment:
```suggestion
In dynamic environments, manual configuration of .maxAllowed can be error-prone, especially as cluster sizes and node configurations change over time. Automating this process by making the VPA recommender aware of the largest node size ensures more reliable scaling decisions. This helps prevent situations where Pods become unschedulable due to excessive resource recommendations.
```

Unfortunately the solution isn't perfect, just because the Pod is slightly smaller than the node, doesn't mean it can be scheduled. 
DaemonSets is a big reason why this could be an issue.

Reply:
Yeah, but we can calculate the node's allocatable size and subtract the DaemonSets from it. that should give us the right value, right?